pkg/storage/stores/shipper/uploads/table.go

```diff
@@ -379,3 +377,10 @@ func loadBoltDBsFromDir(dir string) (map[string]*bbolt.DB, error) {
 
 	return dbs, nil
 }
+
+// getOldestActiveShardTime returns the time of oldest active shard with a buffer of 1 minute.
+func getOldestActiveShardTime() time.Time {
```

Comment:
Admittedly I'm not that familiar with this code, but I'm having a bit of difficulty following. Wouldn't it make more sense as `func getOldestActiveShardTime(from time.Time) time.Time`, which could then be passed `time.Now()` as an argument?

Reply:
We actually always have 1 active shard at a time but we consider last active shard as still active if it was active a minute back. This is to avoid any race conditions. I don't mind changing the function to accept a `time.Time` as a parameter but it would always be getting `time.Now` as input. Do you want me to change it?